crates/containerd-shim-wasmtime/src/executors/container.rs

```diff
@@ -0,0 +1,175 @@
+use libcontainer::workload::{Executor, ExecutorError, EMPTY};
+use nix::unistd::{dup, dup2};
+
+use std::ffi::CString;
+use std::io::Read;
+use std::{fs::OpenOptions, os::fd::RawFd, path::PathBuf};
+
+use libc::{STDERR_FILENO, STDIN_FILENO, STDOUT_FILENO};
+use nix::unistd;
+use oci_spec::runtime::Spec;
+
+const EXECUTOR_NAME: &str = "default";
+
+#[derive(Default)]
+pub struct LinuxContainerExecutor {
+    pub stdin: Option<RawFd>,
+    pub stdout: Option<RawFd>,
+    pub stderr: Option<RawFd>,
+}
+
+impl Executor for LinuxContainerExecutor {
```

Comment:
Is there a reason we can't use the default executor or wrap it?

Reply:
Mostly because I want to implement `can_handle` for the `LinuxContainerExecutor`. The default executor just returns `true` for `can_handle` https://github.com/containers/youki/blob/4dc30bf7c02be584eddfc182d4e96b7cbbb3474c/crates/libcontainer/src/workload/default.rs#L46-L48

Reply:
why not wrap it so we aren't duplicating? I guess I mostly concerned about fixes going into youki like https://github.com/containerd/runwasi/pull/156#discussion_r1284160469 and not getting those in here.  It seems youki should be the place for the main implementation of executing containers not runwasi.

if all we want different is the can_run then again I am wondering if this is the right place for that type of desicion (https://github.com/containerd/runwasi/pull/156#discussion_r1284842502)

Reply:
Changes that go into `verify_binary` should happen at a different layer as where `Executor`'s `can_handle` is invoked. 

> It seems youki should be the place for the main implementation of executing containers not runwasi.

I agree, I can wrap a `DefaultExecutor` instead if the `LinuxContainerExecutor` to reduce duplicate code.

Reply:
> I am wondering if this is the right place for that type of desicion (https://github.com/containerd/runwasi/pull/156#discussion_r1284842502)

I think this totally makes sense, but it can be deferred because it's not a major concern of this PR.

Reply:
Just push a new commit that wraps `DefaultExecutor` instead the `LinuxContainerExecutor`.

Reply:
> > I am wondering if this is the right place for that type of desicion ([#156 (comment)](https://github.com/containerd/runwasi/pull/156#discussion_r1284842502))
> 
> I think this totally makes sense, but it can be deferred because it's not a major concern of this PR.

I think it is though.  the decision making is the core part of the PR and has direct impact on https://github.com/containers/youki/issues/2220

Reply:
I like how the direction is going in https://github.com/containers/youki/issues/2220